src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DocumentationDescriptor.cs

```diff
@@ -0,0 +1,118 @@
+    public static readonly DocumentationDescriptor EventHandlerTagHelper_PreventDefault = new SimpleDescriptor(DocumentationId.EventHandlerTagHelper_PreventDefault);
+    public static readonly DocumentationDescriptor EventHandlerTagHelper_StopPropagation = new SimpleDescriptor(DocumentationId.EventHandlerTagHelper_StopPropagation);
+    public static readonly DocumentationDescriptor KeyTagHelper = new SimpleDescriptor(DocumentationId.KeyTagHelper);
+    public static readonly DocumentationDescriptor RefTagHelper = new SimpleDescriptor(DocumentationId.RefTagHelper);
+    public static readonly DocumentationDescriptor SplatTagHelper = new SimpleDescriptor(DocumentationId.SplatTagHelper);
+
+    public static DocumentationDescriptor Format(DocumentationId id, params object?[] args)
+        => new FormattedDescriptor(id, args);
+
+    public static DocumentationDescriptor From(DocumentationId id, params object?[]? args)
+    {
+        if (args is null or { Length: 0 })
+        {
+            return id switch
+            {
+                DocumentationId.BindTagHelper_Fallback => BindTagHelper_Fallback,
+                DocumentationId.BindTagHelper_Fallback_Event => BindTagHelper_Fallback_Event,
+                DocumentationId.BindTagHelper_Fallback_Format => BindTagHelper_Fallback_Format,
+                DocumentationId.BindTagHelper_Element => BindTagHelper_Element,
+                DocumentationId.BindTagHelper_Element_After => BindTagHelper_Element_After,
+                DocumentationId.BindTagHelper_Element_Culture => BindTagHelper_Element_Culture,
+                DocumentationId.BindTagHelper_Element_Event => BindTagHelper_Element_Event,
+                DocumentationId.BindTagHelper_Element_Format => BindTagHelper_Element_Format,
+                DocumentationId.BindTagHelper_Element_Get => BindTagHelper_Element_Get,
+                DocumentationId.BindTagHelper_Element_Set => BindTagHelper_Element_Set,
+                DocumentationId.BindTagHelper_Component => BindTagHelper_Component,
+                DocumentationId.ChildContentParameterName => ChildContentParameterName,
+                DocumentationId.ChildContentParameterName_TopLevel => ChildContentParameterName_TopLevel,
+                DocumentationId.ComponentTypeParameter => ComponentTypeParameter,
+                DocumentationId.EventHandlerTagHelper => EventHandlerTagHelper,
+                DocumentationId.EventHandlerTagHelper_PreventDefault => EventHandlerTagHelper_PreventDefault,
+                DocumentationId.EventHandlerTagHelper_StopPropagation => EventHandlerTagHelper_StopPropagation,
+                DocumentationId.KeyTagHelper => KeyTagHelper,
+                DocumentationId.RefTagHelper => RefTagHelper,
+                DocumentationId.SplatTagHelper => SplatTagHelper,
+
+                _ => throw new NotSupportedException(Resources.FormatUnknown_documentation_id_0(id))
```

Comment:
I'm assuming the goal of this is preventing any non-cached descriptors from being added when they can be cached? If so, consider adding a comment to let future authors know what they should do if they hit this.

Reply:
Good suggestion. Will do!